message_post_model/model/message_post.py

```diff
@@ -22,273 +22,390 @@
+
+
+def prepare_many2one_info(self, ids, n_obj, field, vals):
+    """Generate the message when many2one fields in the record are changed
+    @param ids: Identifiers of the record to be changed
+    @type ids: integer
+    @param n_obj: Name of the object
+    @type n_obj: str or unicode
+    @param field: name of the field to be changed
+    @type field: str or unicode
+    @param vals: Values of the record to be changed
+    @type vals: dict
+    @return: The message of the changes
+    @rtype: str
+    """
+    obj = self.env[n_obj]
+    message = '<p>'
+
+    last_value = get_last_value(self,
+                                ids, obj._name, field,
+                                obj._fields[field].type)
+    model_obj = self.env[obj._fields[field].comodel_name]
+    model_brw = model_obj.browse(vals[field])
+    new_value = model_brw.name_get()
+    new_value = new_value and new_value[0][1]
+
+    if not (last_value == new_value) and any((new_value, last_value)):
+        message = '<li><b>%s<b>: %s → %s</li>' % \
+            (get_string_by_field(obj, field),
+                get_encode_value(last_value),
+                get_encode_value(new_value))
+    return message
+
+
+def get_encode_value(value):
+    """Encode string values to avoid unicode errors
+    @param value: Any object to try encode the value
+    @type value: str bool date
+    @return: Return decode value
+    @rtype: str
+    """
+    val = value
+    if isinstance(value, (unicode)):
+        val = value.encode('utf-8', 'ignore')
+    return val
+
+
+def prepare_simple_info(self, ids, n_obj, field,
+                        vals):
+    """Generate the message to be shown when simple fields(without
+    relations) are changed
+    @param ids: Identifiers of the record to be changed
+    @type ids: integer
+    @param n_obj: Name of the object
+    @type n_obj: str or unicode
+    @param field: name of the field to be changed
+    @type field: str or unicode
+    @param vals: Values of the record to be changed
+    @type vals: dict
+    @return: The message of the changes
+    @rtype: str
     """
+    obj = self.env[n_obj]
+    message = '<p>'
+    last_value = get_last_value(self,
+                                ids, obj._name, field,
+                                obj._fields[field].type)
+
+    last_value = obj._fields[field].type == 'selection' and \
+        get_selection_value(obj, field, last_value) or last_value
+    new_value = obj._fields[field].type == 'selection' and \
+        get_selection_value(obj, field, vals[field]) or vals[field]
+    last_value = get_encode_value(last_value)
+    new_value = get_encode_value(new_value)
+
+    message = ((last_value != new_value) and
+               any((last_value, vals[field]))) and \
+        '<li><b>%s<b>: %s → %s</li>' % \
+        (get_string_by_field(obj, field), last_value,
+            new_value) or '<p>'
+    return message
 
+
+class MessagePostShowAll(models.Model):
+
+    # TODO: DELETE THIS CLASS
```

Comment:
I don't delete the class, because there are more module with an inheritance over him.

I was testing the concept only with the yoytec module, then this class must be removed and all modules must be changed